invenio_circulation/utils.py

```diff
@@ -168,3 +168,18 @@ def validate_item_pickup_transaction_locations(loan, destination, **kwargs):
     raise NotImplementedConfigurationError(
         config_variable="CIRCULATION_LOAN_LOCATIONS_VALIDATION"
     )
+
+
+def same_location_validator(item_pid, location_pid):
+    """Validates location of given item_pid and given location are the same.
+
+    This also allow for extra validation if needed at the library level.
+
+    :param item_pid: a dict containing `value` and `type` fields to
+        uniquely identify the item.
+    :param location_pid: a location pid.
+    :return: False if validation is not possible, otherwise True
+    """
+    raise NotImplementedConfigurationError(
```

Comment:
Do we want to have a not implemented version by default, or keep the previous one as default?
```
    item_location_pid = current_app.config["CIRCULATION_ITEM_LOCATION_RETRIEVER"](item_pid)
    return input_location_pid == item_location_pid
```

Reply:
done